walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -33,13 +35,48 @@
     Collection<TimeRange> optResults = queryInternal(events, requestWithOptionals);
     
     if ((optResults.isEmpty()) && (!request.getAttendees().isEmpty())) {
-      return queryInternal(events, request);
+      return queryBFS(events, requestWithOptionals, request.getAttendees(),
+          request.getOptionalAttendees());
     }
     else {
       return optResults;
     }
   }
 
+  /* Follow a BFS, removing optional attendees gradually and returning when you find a timerange
+  which accommodates the remaining optional attendees and the required attendees. If you never
+  find such a timerange, return the results just accounting for required attendees. */
+  public Collection<TimeRange> queryBFS(Collection<Event> events, MeetingRequest request, Collection<String> reqAttendees,
```

Comment:
Here I suggest just passing in request.duration() since I think that's the only thing we need from request right? Makes it clearer.

Reply:
For sure! Thank you